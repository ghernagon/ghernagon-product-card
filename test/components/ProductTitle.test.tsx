import React from 'react';
import renderer from 'react-test-renderer';
import {ProductTitle, ProductCard} from '../../src/components';
import { product } from '../data/products';


describe('Test in <ProductTitle />', () => {
    it('should render component with custom title', () => {
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product Title"/>
        )

        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should show component with original title', () => {
        const wrapper = renderer.create(
                <ProductCard product={product}>
                    {
                        () => (
                                <ProductTitle />
                        )
                    }
                </ProductCard>
        );

        expect(wrapper.toJSON()).toMatchSnapshot();
    });
});