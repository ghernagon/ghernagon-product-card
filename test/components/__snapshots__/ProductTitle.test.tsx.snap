// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ProductTitle /> should render component with custom title 1`] = `
<span
  className="productDescription undefined"
>
  Custom Product Title
</span>
`;

exports[`Test in <ProductTitle /> should show component with original title 1`] = `
<div
  className="productCard undefined"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Card
  </span>
</div>
`;
